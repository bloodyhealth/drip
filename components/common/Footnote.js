import React from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, View } from 'react-native'

import AppText from '../common/app-text'
import Asterisk from '../common/Asterisk'

import { Colors, Spacing } from '../../styles'

const Footnote = ({ children, colorLabel }) => {
  if (!children) return false

  return (
    <View style={styles.container}>
      <Asterisk />
      <AppText
        linkStyle={styles.link}
        style={{ ...styles.text, color: Colors[colorLabel] }}
      >
        {children}
      </AppText>
    </View>
  )
}

Footnote.propTypes = {
  children: PropTypes.node,
  colorLabel: PropTypes.string,
}

Footnote.defaultProps = {
  colorLabel: 'greyDark',
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignContent: 'flex-start',
    marginBottom: Spacing.tiny,
    marginTop: Spacing.base,
  },
  link: {
    color: 'white',
  },
  text: {
    paddingLeft: Spacing.small,
  },
})

export default Footnote
