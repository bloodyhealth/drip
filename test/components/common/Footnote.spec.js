import React from 'react'
import { render } from '@testing-library/react-native'

import Footnote from '../../../components/common/Footnote'

describe('Footnote component', () => {
  test('when children are present, renders them', () => {
    const text = 'Some footnote text'
    const { toJSON } = render(<Footnote>{text}</Footnote>)

    expect(toJSON()).toMatchSnapshot()
  })

  test('when no children, renders nothing', () => {
    const { toJSON } = render(<Footnote></Footnote>)

    expect(toJSON()).toMatchSnapshot()
  })
})
