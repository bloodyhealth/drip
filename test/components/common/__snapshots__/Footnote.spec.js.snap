// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footnote component when children are present, renders them 1`] = `
<View
  style={
    Object {
      "alignContent": "flex-start",
      "flexDirection": "row",
      "marginBottom": 8.571428571428571,
      "marginTop": 21.428571428571427,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#555",
          "fontFamily": "Jost-Book",
          "fontSize": 34.285714285714285,
        },
        Object {
          "color": "#F38337",
        },
      ]
    }
  >
    *
  </Text>
  <Text
    linkStyle={
      Object {
        "color": "white",
      }
    }
    style={
      Array [
        Object {
          "color": "#555",
          "fontFamily": "Jost-Book",
          "fontSize": 34.285714285714285,
        },
        Object {
          "color": "#CCC",
          "paddingLeft": 34.285714285714285,
        },
      ]
    }
  >
    Some footnote text
  </Text>
</View>
`;

exports[`Footnote component when no children, renders nothing 1`] = `null`;
